CSharpMath/Atom/Atoms/Comment.cs

```diff
@@ -0,0 +1,8 @@
+namespace CSharpMath.Atom.Atoms {
+  public sealed class Comment : MathAtom {
```

Comment:
What is a comment MathAtom? Needs documentation.

Reply:
We should keep Comments when e.g. converting LaTeX to MathML and vice versa.

Reply:
Are comments part of the visual structure of a mathematical expression? If not then they shouldn't be in MathAtom.

Reply:
They are not part of the visual structure. Visual structure is represented by `Display`s and `MathAtom`s are never meant to be the visual structure.

Reply:
Why should comments be dropped in LaTeX to MathML conversion?

Reply:
1. If the definition of a MathAtom is a representation of the visual structure of a mathematical expression, an invisible comment doesn't form part of that. 2. If the definition were changed to add "together with invisible comments positioned at specific locations" it would still not be able to add latex comments because latex comments are comments at positions in the latex and that doesn't map to positions in the expression structure.

The comments are comments on the expression not part of it.

Reply:
How would you define a MathAtom otherwise?

Reply:
>If the definition of a MathAtom is a representation of the visual structure of a mathematical expression, an invisible comment doesn't form part of that

Again, those are called `Display`s. `Display`s are visual structures in that they have visual data such as positioning and typographic measures.
>If the definition were changed to add "together with invisible comments positioned at specific locations" it would still not be able to add latex comments because latex comments are comments at positions in the latex and that doesn't map to positions in the expression structure.

There are no "positioned at specific locations". They do not have a position. (`Display`s do.)
>The comments are comments on the expression not part of it.

Comments commentate the expression and should not be dropped in converting MathML <-> LaTeX.
>How would you define a MathAtom otherwise?

I think of `MathAtom`s more like a syntax tree.

Reply:
> Display

Visual appearance (represented by Display) has more detail than the visual structure...

> There are no "positioned at specific locations". They do not have a position. (Displays do.)

Everything in the atom has a position in the tree. I don't think comments can have such a position.

> a syntax tree simplified from LaTeX

This is an inferior definition to "the structure of a mathematical expression" since it just defers the definition to LaTeX. It also doesn't justify inclusion of comments because it's not clear that comments are part any intrinsic LaTeX syntax tree.

Reply:
>Visual appearance (represented by Display) has more detail than the visual structure...

I'd define visual appearance as the drawn output, not Displays. That is because there are still various settings like default text color etc. when drawing Displays. Therefore, Displays only represent the visual structure.
>Everything in the atom has a position in the tree. I don't think comments can have such a position.

The position in the tree can be derived from where you write the comment.
e.g.
```tex
a_0 % The initial term of the sequence 
+ a_1
```
is different from
```tex
a_0
+ a_1 % The initial term of the sequence
```
.
>This is an inferior definition to "the structure of a mathematical expression" since it just defers the definition to LaTeX. 

In "the structure of a mathematical expression", there is also ambiguity on what is a "mathematical expression". In Mathematics, you do not care about what color the equations are. However, we have `Colored` as a `MathAtom` because we have `\color` in LaTeX to change equation colors. 
The definition "the structure of a mathematical expression" suits `CSharpMath.Evaluation.MathItem` more, because it can only represent meaningful math operations.
>It also doesn't justify inclusion of comments because it's not clear that comments are part any intrinsic LaTeX syntax tree.

Syntax trees correspond to source code. You can take a look at the [Roslyn syntax tree](https://sharplab.io/#v2:C4LghgzsA+ACBMBGAsAKFgZgAQKwYSwG80tSdtYAWLAWQAoBKIks1gejfwHsBbHgUwB2wAIQiWpAL5pJQA==) includes trivia such as comments. Our `MathAtom` is simplified from the original LaTeX in that unlike syntax trees, we expand macros and do not keep the original commands.

Reply:
LaTeX is a WYSISYM approach to structuring visual textual and mathematical content. So the mathematical part is a visual structure of mathematical expression. Note the word "a": other structures are possible, including strucures that preserve the exact visual output, whereas an Atom, like LaTeX/MathML will preserve only some visual structure, enough to preserve meaning but not enough to preserve complete rendering accuracy. Color is there because a visual structure of an expression is captured, not a semantic structure (as content MathML for example captures).

Reply:
And comments have meaning. These meanings are dropped during LaTeX <-> MathML conversion if we don't preserve comments.